MON-157180-v3-gorgone-cloud-poller-disconnect-when-gorgone-log-are-to-big-to-be-sent (#2571)
centreon/centreon-collect: gorgone/docs/modules/core/pullwss-log-sync.md

## Gorgone log synchronization with pullwss communication mode.

> each box represent a process running. Each participant represent a file used by the process.\

Reviewer: ```suggestion
Each box represents a process running. Each participant represents a file used by the process.\
```

> as participant name must be unique, P or C represent if the file is used on the Poller or the Central\

Reviewer: ```suggestion

As participants' names must be unique, P or C represents whether the file is used on the Poller or the Central\
```

> dotted line are zmq message, full line are direct subroutine call.

Reviewer: ```suggestion

The dotted lines are zmq messages, the full lines are direct subroutine calls
```

> ### Pullwss

Reviewer: ```suggestion
### Pullwss

```

> Please note that pullwss communication mode have a limitation on the message size that zmq don't have.

Reviewer: ```suggestion
Please note that the pullwss communication mode has a limitation on message size that zmq doesn't have.
```

> To limit the size of the GETLOG message, the pullwss module split the message in smaller part, and send multiple SETLOGS message.

Reviewer: ```suggestion
To limit the size of the GETLOG message, the pullwss module splits the message into smaller parts, and sends multiple SETLOGS messages.
```

> Other communication module send only one SETLOGS message in this case.

Reviewer: ```suggestion
Other communication modules send only one SETLOGS message in this case.
```

    end
    box rgb(100,100,102) central-proxy-httpserver
        participant C/class/module
        participant C/proxy/httpserver
    end

    box rgb(100,100,102) poller-pullwss
        participant P/pullwss/class
        participant P/class/module
    end
    box rgb(100,100,102) poller-core
        participant P/class/core
        participant P/library
    end

    C/class/core -> + C/class/core: run
    C/class/core -> + C/class/core: EV::timer(every 5s)
    C/class/core -> + C/class/core: periodic_exec
    C/class/core -> + C/class/core: check_exit_modules
    C/class/core ->> + C/proxy/hook: check
    C/proxy/hook -> + C/proxy/hook: full_sync_history
    note left of C/proxy/hook: full_sync_history is called<br/> every synchistory_time <br/>sec, see proxy.md
    C/proxy/hook -> + C/proxy/hook: routing
    C/proxy/hook -> C/proxy/hook: get_sync_time
    C/proxy/hook -> C/proxy/hook: send_internal_message
    C/proxy/hook --) + C/proxy/httpserver: GETLOG
    deactivate C/proxy/hook
    deactivate C/proxy/hook
    deactivate C/proxy/hook
    deactivate C/class/core
    deactivate C/class/core
    deactivate C/class/core
    deactivate C/class/core

> # central use the proxy module to send data to the remote node. proxy have multiples process runing in parallel,

Reviewer: ```suggestion
# The central uses the proxy module to send data to the remote node. Proxies have multiples processes runing in parallel,
```

> # it is not clear for now if the main process see message before the worker process.

Reviewer: ```suggestion
# it is not clear for now if the main process sees the message before the worker process.
```

    C/proxy/httpserver ->  + C/proxy/httpserver: read_zmq_events
    C/proxy/httpserver ->  + C/proxy/httpserver: proxy
    C/proxy/httpserver --) P/pullwss/class: GETLOGS
    deactivate C/proxy/httpserver
    deactivate C/proxy/httpserver

> # the poller (which is a distant node) retrieve the GETLOG message, and process it.

Reviewer: ```suggestion
# The poller (which is a distant node) retrieves the GETLOG message, and processes it.
```

> # first the pullwss module listen the websocket, and transmit the message retrieved to the core.

Reviewer: ```suggestion
# First the pullwss module listens on the websocket, and transmits the message retrieved to the core.
```

> # the core process it and send back the response.

Reviewer: ```suggestion
# The core processes it and sends back the response.
```

> # TODO : make a note wss_connect is not called everytime.

Reviewer: ```suggestion
# TODO : make a note that wss_connect is not called every time.
```

    P/pullwss/class -> + P/pullwss/class: wss_connect
    P/pullwss/class ->> + P/class/module: send_internal_action
    P/class/module --) P/class/core: GETLOG
    P/class/core -> + P/class/core: router_internal_event
    P/class/core -> + P/class/core: message_run
    P/class/core ->> P/library: getlog
    P/class/core ->  + P/class/core: send_internal_response
    P/class/core --) P/pullwss/class: GETLOG
    deactivate P/class/core
    deactivate P/class/core
    deactivate P/class/core

>     # on the distant poller, processing of the message.

Reviewer: ```suggestion
    # On the distant poller, processing of the message.
```

    P/pullwss/class -> + P/pullwss/class: read_zmq_events
    P/pullwss/class -> + P/pullwss/class: transmit_back

>     loop split the logs in smaller message if it's too big for the websocket, making possibility multiples SETLOG message

Reviewer: ```suggestion
    A loop splits the logs into smaller messages if they're too big for the websocket, possibly making multiple SETLOG messages
```

Author: loop is a reserved word to make graph with mermaid, I'll do this on the next commit
